deeprank/features/Edesolv.py

```diff
@@ -0,0 +1,411 @@
+import warnings
+
+from pdb2sql import pdb2sql, interface
+from deeprank.features import FeatureClass
+from copy import deepcopy
+from io import StringIO
+import numpy as np
+
+try:
+    from Bio.PDB import PDBParser, SASA
+
+except ImportError:
+    warnings.warn('Biopython module not found. Make sure you have Biopython 1.79 or later')
+    raise
+
+
+class Edesolv(FeatureClass):
+
+    # init the class
+    def __init__(self, pdb_data):
+        """Compute the energy of desolvation feature.
+
+        Biopython 1.79 or later is required for this feature.
+
+        Args:
+            pdb_data (list(byte) or str): pdb data
+
+        Example:
+            >>> edesolv = Edesolv('1AK4.pdb')
+            >>> edesolv.get_feature()
+        """
+        self.pdb = pdb_data
+        print('PDB DATA: ')
+        print(pdb_data)
+        print('TYPE: ' + str(type(pdb_data)))
+
+
+        self.feature_data = {}
+        self.feature_data_xyz = {}
+    
+    # the feature extractor
+    def get_feature(self, verbose=False):
+        """ Computes the actual Edesolv feature"""
+        
+        esolcpx = 0.0
+        esolfree = 0.0
+        
+        # create a sql database
+        pdb_db = pdb2sql(self.pdb)
+        self.db = interface(pdb_db)
+        
+        #Create a temporary pdb-like string to calculate complex and free Edesolv
+        temp_pdb = StringIO(('\n').join(pdb_db.sql2pdb()))
+        temp_pdb.seek(0)
+
+        # Parse structure in Bio.PDB
+        p = PDBParser(QUIET=1) 
+        struct = p.get_structure('temp', temp_pdb)
+        
+        # Identify chains in the Bio.PDB object
```

Comment:
It's needed to consider multi-chain complexes and various chain names (it might not be A and B). Ffor that, you'll have to ask users to provide the chainIDs for chain1 and chain2. For detail see how it's implemented for features.

Reply:
Fixed in 6fed0fc